src/control/cmd/daos_agent/fabric.go

```diff
@@ -247,20 +248,31 @@ func (n *NUMAFabric) getNextDevice(numaNode int) *FabricInterface {
 }
 
 func (n *NUMAFabric) findOnAnyNUMA(netDevClass uint32, provider string) (*FabricInterface, error) {
-	numNodes := n.getNumNUMANodes()
+	nodes := n.getNUMANodes()
+	numNodes := len(nodes)
+
 	for i := 0; i < numNodes; i++ {
-		numa := (n.defaultNumaNode + i) % numNodes
-		fi, err := n.getDeviceFromNUMA(numa, netDevClass, provider)
+		n.currentNUMANode = (n.currentNUMANode + 1) % numNodes
+		fi, err := n.getDeviceFromNUMA(nodes[n.currentNUMANode], netDevClass, provider)
 		if err == nil {
 			// Start the search on this node next time
-			n.defaultNumaNode = numa
-			n.log.Debugf("Suitable fabric interface %q found on NUMA node %d", fi.Name, numa)
+			n.log.Debugf("Suitable fabric interface %q found on NUMA node %d", fi.Name,
+				nodes[n.currentNUMANode])
```

Comment:
Strange that there is not the same value from the original patch. From what I understand, the debug message from the patch is probably invalid: `n.currentNUMANode` replace in this patch with `nodes[n.currentNUMANode]`.

Reply:
I think I forgot to change it in the original patch. I had to go in manually on this one to resolve conflicts and ended up noticing this.